src/main/java/com/mewna/catnip/entity/builder/PresenceBuilder.java

```diff
@@ -0,0 +1,53 @@
+package com.mewna.catnip.entity.builder;
+
+import com.mewna.catnip.entity.impl.PresenceImpl;
+import com.mewna.catnip.entity.impl.PresenceImpl.ActivityImpl;
+import com.mewna.catnip.entity.user.Presence;
+import com.mewna.catnip.entity.user.Presence.Activity;
+import com.mewna.catnip.entity.user.Presence.ActivityType;
+import com.mewna.catnip.entity.user.Presence.OnlineStatus;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+import lombok.experimental.Accessors;
+
+import javax.annotation.CheckReturnValue;
+import javax.annotation.Nonnull;
+
+/**
+ * @author SamOphis
+ * @since 10/12/2018
+ */
+@Setter(onParam_ = @Nonnull, onMethod_ = {@CheckReturnValue, @Nonnull}) // doesn't work in my ide, but *should* work elsewhere
```

Comment:
We don't do this anywhere else, so not sure if we should remove it here or add it everywhere else.

Reply:
we should probably add it everywhere else simply to keep consistency with the rest of the public API